drivers/gpio/gpio_brcmstb.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2024 Junho Lee <junho@tsnlab.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT brcm_brcmstb_gpio
+
+#include <zephyr/arch/common/sys_bitops.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+#define GIO_DATA  0x04
+#define GIO_IODIR 0x08
+
+#define DEV_CFG(dev)  ((const struct gpio_brcmstb_config *)(dev)->config)
+#define DEV_DATA(dev) ((struct gpio_brcmstb_data *)(dev)->data)
+
+struct gpio_brcmstb_config {
+	struct gpio_driver_config common;
+
+	DEVICE_MMIO_NAMED_ROM(reg_base);
+	mem_addr_t offset;
+};
+
+struct gpio_brcmstb_data {
+	struct gpio_driver_data common;
+
+	DEVICE_MMIO_NAMED_RAM(reg_base);
+	mem_addr_t base;
+};
+
+static int gpio_brcmstb_pin_configure(const struct device *port, gpio_pin_t pin, gpio_flags_t flags)
+{
+	struct gpio_brcmstb_data *data = port->data;
```

Comment:
Null check?

Reply:
First it's not my patch and it's explicitly marked as back-ported (with "cherry picked from commit 31baddaa51d5b40e6e6394b8be6cb32e1e962f08", so probably right place for comments is in Zephyr mainline https://github.com/zephyrproject-rtos/zephyr/pull/70538.

Second, those functions are not called directly, but through the gpio APIs and null is impossible here.